crates/lib3h/src/dht/mirror_dht.rs

```diff
@@ -301,18 +282,46 @@ impl MirrorDht {
             peer_address_list: self.get_other_peer_list(),
             bundle: buf,
         };
-        DhtEvent::GossipTo(gossip_evt)
+        DhtRequestToParent::GossipTo(gossip_evt)
     }
+}
+
+/// Impl DhtActor interface
+impl
+    GhostActor<
+        DhtRequestToParent,
+        DhtRequestToParentResponse,
+        DhtRequestToChild,
+        DhtRequestToChildResponse,
+        Lib3hError,
+    > for MirrorDht
+{
+    fn take_parent_endpoint(&mut self) -> Option<DhtEndpoint> {
+        std::mem::replace(&mut self.endpoint_parent, None)
+    }
+
+    fn process_concrete(&mut self) -> GhostResult<WorkWasDone> {
+        detach_run!(&mut self.endpoint_self, |es| es.process(&mut ()))?;
+        for request in self.endpoint_self.as_mut().drain_messages() {
+            //debug!("@MirrorDht@ serving request: {:?}", request);
+            self.handle_request_from_parent(request)
+                .expect("no ghost errors");
+        }
+        let (did_work, command_list) = self.internal_process().unwrap(); // FIXME unwrap
+        for command in command_list {
+            self.endpoint_self.publish(command);
+        }
+        Ok(did_work.into())
+    }
+}
 
-    /// Process a DhtEvent Command, sent by our owner.
-    /// Return a list of DhtEvent to owner.
-    #[allow(non_snake_case)]
-    fn serve_DhtCommand(&mut self, cmd: &DhtCommand) -> Lib3hResult<Vec<DhtEvent>> {
-        debug!("@MirrorDht@ serving cmd: {:?}", cmd);
-        // Note: use same order as the enum
-        match cmd {
+impl MirrorDht {
+    #[allow(irrefutable_let_patterns)]
+    fn handle_request_from_parent(&mut self, mut request: DhtToChildMessage) -> Lib3hResult<()> {
+        debug!("@MirrorDht@ serving request: {:?}", request);
+        match request.take_message().expect("exists") {
```

Comment:
I think this api is in flux, but if it compiles now, then it's good enough for me.